packages/cli/generators/datasource/index.js

```diff
@@ -150,11 +150,13 @@ module.exports = class DataSourceGenerator extends ArtifactGenerator {
   promptConnectorConfig() {
     debug('prompting for connector config');
     // Check to make sure connector is from connectors list (not custom)
-    const settings = connectors[this.artifactInfo.connector]
-      ? connectors[this.artifactInfo.connector]['settings']
-      : {};
-
+    const settings =
+      connectors[this.artifactInfo.connector] &&
+      connectors[this.artifactInfo.connector]['settings']
+        ? connectors[this.artifactInfo.connector]['settings']
+        : {};
```

Comment:
It can be simplified as:
```ts
const settings =
      (connectors[this.artifactInfo.connector] &&
      connectors[this.artifactInfo.connector]['settings']) || {};
```

Reply:
good example of using javascript sintax to simplify it. 👍